Added background music and managed card logic

MeshiSanker opened the pull request.

MeshiSanker committed:
Now you can run another game and keep playing after tie:
	- Added "Start a New Game" button to the winner view, will start a new match after clicking it.
	- Now when there is a tie each player will get 5 cards and the game will continue utill there is a winner.
	- Improved card dividing method.

MeshiSanker committed:
Added background music to each view.

MeshiSanker committed:
Fix

OmriGilhar commented on .idea/gradle.xml:
```diff
@@ -7,7 +7,6 @@
         <option name="testRunner" value="PLATFORM" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="1.8 (2)" />
```
?? What is this ??

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Game_Activity.java:
```diff
@@ -45,6 +47,8 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.game_activity);
+        backgroundSong = MediaPlayer.create(Game_Activity.this, R.raw.loyalty_freak_music10the_witch_are_going_magical);
```
Move it to init views, please don't make the onCreate dirty

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Game_Activity.java:
```diff
@@ -203,9 +208,16 @@ private void refreshScoreView() {
         int card_value;
         int card_shape;
+        int i=0;
         boolean player_card_flag = true;
```
Remove this variable, you remove the logic regarding this, please delete it.

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Game_Activity.java:
```diff
@@ -223,15 +235,19 @@ private void createGameStacks() {
-            }
+        // In case of tie, give only 5 cards.
+        while(all_card_stack.size() != 0 && i < 5){
+            // Add card to player one stack and remove it from overall stack
```
// Add card to player one stack and remove it from the overall stack
Change the overall stack --> list

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Game_Activity.java:
```diff
@@ -223,15 +235,19 @@ private void createGameStacks() {
-                insertCardToPlayerTwoStack(all_card_stack.get(0));
-                player_card_flag = true;
-            }
+        // In case of tie, give only 5 cards.
```
You need to explain why we can use `all_card_stack.remove(0);` (always even cards)

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Menu_Activity.java:
```diff
@@ -3,18 +3,22 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.menu_activity);
+        backgroundSong = MediaPlayer.create(Menu_Activity.this, R.raw.loyalty_freak_music04hello_regan);
```
Again move it to init view ( we want to leave the onCreate the cleanest) personally I checked it, and he is right, the best practice is to leave this method the cleanest.

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Winner_Activity.java:
```diff
@@ -14,14 +17,17 @@
         this.winner_score = getIntent().getIntExtra("winner_score", -1);
         this.winner_img_id = getIntent().getIntExtra("winner_image_id", -1);
-
+        backgroundSong = MediaPlayer.create(Winner_Activity.this, R.raw.loyalty_freak_music04hello_regan);
```
Again =D move to init views

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Winner_Activity.java:
```diff
@@ -40,6 +47,15 @@ private void initViews() {
 
+    private void openGameView() {
+        backgroundSong.release();
+        Log.println(Log.DEBUG, "menu", "Open game");
```
Remove debag call

OmriGilhar commented on app/src/main/java/com/example/msogcardwar/Winner_Activity.java:
```diff
@@ -40,6 +47,15 @@ private void initViews() {
+        Intent gameIntent= new Intent(Winner_Activity.this, Game_Activity.class);
+        Winner_Activity.this.startActivity(gameIntent);
     }
 }
```
Add a newline at the end of every file

OmriGilhar commented on app/src/main/res/layout/winner_activity.xml:
```diff
@@ -36,6 +36,15 @@
             android:textColor="@color/white"
             android:textSize="30sp"/>
+        <Button
+            android:id="@+id/startNew_Game_BTN"
```
The naming convention is

<activity_name>\_<BTN/IMG/LBL....>\_<component_name>
In this case please change it to:

winner_BTN_startNewGame

OmriGilhar commented on app/src/main/res/layout/winner_activity.xml:
```diff
@@ -36,6 +36,15 @@
+            android:textColor="@color/black"/>
     </LinearLayout>
 
 </androidx.constraintlayout.widget.ConstraintLayout>
```
Add a newline at the end of every file

OmriGilhar commented on app/src/main/res/values/strings.xml:
```diff
@@ -13,6 +13,7 @@
 
     <string name="winner_title">!! WINNER !!</string>
     <string name="winner_score_title">Your Score </string>
+    <string name="start_new_game">Start a New Game </string>
```
The naming convention is:

<activity_related>_<string_title>
in this case:

winner_start_new_game

OmriGilhar commented on app/src/main/res/layout/winner_activity.xml:
```diff
@@ -36,6 +36,15 @@
+            android:id="@+id/startNew_Game_BTN"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/start_new_game"
```
You can add a not all upper case option, look at the menu activity --> start button, 
look at the configuration of the button in the XML and you will find an option to let the string use case sensitive.

OmriGilhar reviewed (changes requested):
Hey Mesh,

Please take a look at the comments I made, fix it, and send a new pull request =D

MeshiSanker committed:
Fixed last commit issues and added onDestroy methods to each viewer.
	- Now when call to onDestroy the music will stop and the right viewer will appear.

Omri-Gilhar committed:
Removed music, Move the logic outside from acitvity.

	* Game Manager:
		* Manage the game from start to winner_activity.
		* Contain 2 players, and winner.
	* Player:
		* Contain the stack, score, and current card.
